### PR TITLE
TC 2023-05-31

### PR DESCRIPTION
**Pull request by HeikoTheissen**
- [x] Give example data and response for the "hierarchy switching" example 109. Explain why `traverse` cannot be omitted.

### CODE REVIEW

File: odata-data-aggregation-ext/5 Vocabulary for Data Aggregation.md [hunk 1]
```diff
@@ -161,19 +157,19 @@ A _leveled hierarchy_ has a fixed number of levels each of which is represented
 
 A leveled hierarchy can be defined for a collection of instances of an entity or complex type and is described with the term `LeveledHierarchy` that lists the properties used to form the hierarchy.
 
-The order of the collection is significant: it lists the properties representing the levels, starting with the root level (coarsest granularity) down to the lowest level of the hierarchy. A leveled hierarchy is defined by an ordered list of groupable properties of the entity or complex type that represent the root level and the consecutively finer-grained levels of the hierarchy.
+The order of the collection is significant: it lists the properties representing the levels, starting with the root level (coarsest granularity) down to the lowest (finest-grained) level of the hierarchy.
```
Comment (HeikoTheissen) [inline odata-data-aggregation-ext/5 Vocabulary for Data Aggregation.md:160]: @GeraldKrause, please review

File: odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md
Comment (HeikoTheissen) [inline odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md:1]: Given `Amount`s -1, 0 and 1, both `bottomsum(-2,Amount)` and `bottomsum(2,Amount)` return all three. Is this really intended?

Comment (HeikoTheissen) [inline odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md:1]: Why do `identity` and `compute` preserve the order?

File: odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md [hunk 2]
```diff
@@ -1,31 +1,31 @@
 ## ##subsec Transformations Preserving the Input Set Structure
```
Comment (ralfhandl) [inline odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md:1]: Why not name the section "Transformations Producing a Subset of the Input Set"?

Comment (HeikoTheissen) [reply odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md:1]: Discuss in TC. For me the more important thing is that the structure is preserved.

File: odata-data-aggregation-ext/6 Hierarchical Transformations.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ The parameter lists defined in the following subsections have three mandatory pa
...
 
 The recursive hierarchy to be processed can also be a subset $H'$ of $H$. For this case a non-empty sequence $S$ of transformations MAY be specified as an optional parameter whose position varies from transformation to transformation and is given below. In general, let $H'$ be the output set of the transformation sequence $S$ applied to $H$, or $H'=H$ if $S$ is not specified. The transformations in $S$ MUST be listed in the section on [Transformations Preserving the Input Set Structure](#TransformationsPreservingtheInputSetStructure) or in the section on [Hierarchical Transformations Preserving the Input Set Structure](#HierarchicalTransformationsPreservingtheInputSetStructure) or be service-defined bound functions whose output set is a subset of the input set.
```
Comment (ralfhandl) [inline odata-data-aggregation-ext/6 Hierarchical Transformations.md:23]: Why not rename both referenced sections to "(Hierarchical) Transformations Producing a Subset of the Input Set"?

And remove the first sentence explaining that "preserve structure" actually means "subset".

File: odata-data-aggregation-ext/6 Hierarchical Transformations.md [hunk 4]
```diff
@@ -18,7 +18,7 @@ The parameter lists defined in the following subsections have three mandatory pa
...
```
Comment (ralfhandl) [inline odata-data-aggregation-ext/6 Hierarchical Transformations.md:24]: Regarding bound functions that serve as additional hierarchical transformations: are they also required to produce a subset of the input set, or can they for example invent transient entities with the input set structure out of thin air?

Comment (ralfhandl) [reply odata-data-aggregation-ext/6 Hierarchical Transformations.md:24]: Apparently not, the last part of the preceding paragraph also restricts them to subsetting functions:

> or be service-defined bound functions whose output set is a subset of the input set.

Comment (HeikoTheissen) [reply odata-data-aggregation-ext/6 Hierarchical Transformations.md:24]: `traverse` preserves the input set structure but duplicates nodes with multiple parents, therefore it does not produce a subset.

File: odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md
Comment (HeikoTheissen) [inline odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md:1]: > The transformation orderby defined below carries out a stable-sort by a list of expressions. This is a sorting operation on an ordered collection

This reads as if `orderby` cannot be performed on an unordered collection.

File: odata-data-aggregation-ext/3.1 Fundamentals of Input and Output Sets.md
Comment (HeikoTheissen) [inline odata-data-aggregation-ext/3.1 Fundamentals of Input and Output Sets.md:1]: > But when such a collection is transformed by the `$apply` system query option, **halfway** cases between ordered and unordered can arise.

Native speakers?

File: odata-data-aggregation-ext/5 Vocabulary for Data Aggregation.md (hunk 1 above)
Comment (GeraldKrause) [reply odata-data-aggregation-ext/5 Vocabulary for Data Aggregation.md:160]:
```suggestion
The order of the collection is significant: it lists paths from the entity or complex type where the term is applied to groupable properties representing the levels, starting with the root level (coarsest granularity) down to the lowest (finest-grained) level of the hierarchy.
```

File: odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md
Comment (HeikoTheissen) [reply odata-data-aggregation-ext/3.3 Transformations Preserving the Input Set Structure.md:1]: Aligned with MDX.

**Pull request merged**